runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java

```diff
@@ -144,6 +170,20 @@ public void onChatMessage(ChatMessage chatMessage)
 					notifier.notify(chatMessage.getMessage());
 				}
 				break;
+			case BROADCAST:
+				if (chatMessage.getMessage().contains("|") && config.notifyOnBroadcast())
+				{
+					// Broadcasts can accompanied by a clickable link that is notated by `|` followed by a number that determines which website is linked.
+					// This prevents it from printing in broadcast notifications.
+					String broadcast = removeTags(chatMessage.getMessage());
+					notifier.notify(broadcast.substring(0, broadcast.lastIndexOf('|')));
+				}
+				if (config.notifyOnBroadcast())
+				{
+					String broadcast = removeTags(chatMessage.getMessage());
+					notifier.notify(broadcast);
+				}
```

Comment:
This would notify twice on broadcasts that have links. I recommend combining the bodies of these two conditional blocks by doing the following: make a guard clause for `config.notifyOnBroadcast()`, then a check for `broadcast.lastIndexOf('|')` being `-1` (if it isn't, get the substring prior to that vertical bar). Finally, remove the formatting tags, then notify that string.